src/FSharp.DependencyManager.Nuget/FSharp.DependencyManager.fs

```diff
@@ -24,12 +24,22 @@ module FSharpDependencyManager =
         | true, false -> v
         | _ -> ""
 
-    let validateAndFormatRestoreSources (sources: string) =
+    let validateAndFormatRestoreSources (directoryPath: string) (sources: string) =
         [|
             let items = sources.Split(';')
 
             for item in items do
-                let uri = Uri(item)
+                let uri =
+                    let mutable res: Uri = null
+                    if Uri.TryCreate(item, UriKind.Absolute, &res) then
+                        res
+                    else
+                        let baseUri =
+                            if directoryPath.EndsWith @"\" || directoryPath.EndsWith "/" then
+                                Uri(directoryPath)
+                            else
+                                Uri(directoryPath + "/")
```

Comment:
This doesn't feel right. Doesn't dotnet have proper APIs, which allow you to construct paths in a way, which is system independent?

Reply:
Yeah, you are right, I was confused by Uri and UriBuilder, neither of which handles it, but there is System.IO.Path.Combine, I will use that.